File: examples/football.cs
using System;
using System.Text;
using System.Collections.Generic;
using System.Linq;
using System.IO;

public class Program
{
    static List<Tuple<string, string, int[]>> LoadResults(string filename)
    {
        var raw = File.ReadAllText(filename);
        var lines = raw.Split('\n');
        return lines.Where(
            line => line.Length != 0
        ).Select(
            line => ParseResult(line)
        ).ToList();
    }

static Tuple<string, string, int[]> ParseResult(string line)
    {
        var awayIndex = line.IndexOf(" - ") + 3;
        var resultIndex = line.IndexOf(" ", awayIndex) + 1;
        var goals = line.Substring(resultIndex).Split(':');
        return Tuple.Create(
            line.Substring(0, awayIndex - 3),
            line.Substring(awayIndex, resultIndex - 1 - awayIndex),
            new[] { Int32.Parse(goals[0]), Int32.Parse(goals[1]) }
        );
    }

static int CalculatePoints(List<Tuple<string, string, int[]>> results, string team)
    {
        return results.Aggregate(0, (memo, result) => memo + ResultPoints(team, result.Item1, result.Item2, result.Item3));
    }

static int ResultPoints(string team, string host, string away, int[] result)
    {
        if (host == team && result[0] > result[1] || away == team && result[0] < result[1])
        {
            return 3;
        }
        else if (result[0] == result[1] && (host == team || away == team))
        {
            return 1;
        }
        else 
        {
            return 0;
        }
    }

    public static void Main(string[] args)
    {
        if (args.Length < 2)
        {
            Console.WriteLine("usage: football <stats-file> <team>");
        }
        else 
        {
            var results = LoadResults(args[0]);
            Console.WriteLine(CalculatePoints(results, args[1]));
        }
    }
}

File: examples/football.py
import sys
import functools

def load_results(filename):
    with open(filename, 'r') as _f:
        raw = _f.read()

    lines = raw.split('\n')
    return [parse_result(line) for line in lines if line]

def parse_result(line):
    away_index = line.index(' - ') + 3
    result_index = line.index(' ', away_index) + 1
    goals = line[result_index:].split(':')
    return (line[:away_index - 3], line[away_index:result_index - 1], (int(goals[0]), int(goals[1])))

def calculate_points(results,team):
    return functools.reduce(lambda memo, result: memo + result_points(team, result[0], result[1], result[2]), results, 0)

def result_points(team,host,away,result):
    if host == team and result[0] > result[1] or away == team and result[0] < result[1]:
        return 3
    elif result[0] == result[1] and (host == team or away == team):
        return 1
    else:
        return 0

if len(sys.argv) < 3:
    print('usage: football <stats-file> <team>')
else:
    results = load_results(sys.argv[1])
    print(calculate_points(results, sys.argv[2]))

